runtime/src/policy.rs

```diff
@@ -0,0 +1,206 @@
+            consensus_verifier,
+            logger,
+        }
+    }
+
+    /// Fetch runtime's quote policy from the latest verified consensus layer state.
+    ///
+    /// If the runtime version is not provided, the policy for the last known epoch is returned.
+    pub fn quote_policy(
+        &self,
+        ctx: Arc<Context>,
+        runtime_id: &Namespace,
+        version: Option<Version>,
+        use_latest_state: bool,
+    ) -> Result<QuotePolicy> {
+        // Verify to the latest height, if needed.
+        let consensus_state = if use_latest_state {
+            self.consensus_verifier.latest_state()?
+        } else {
+            self.consensus_verifier.state_at(HEIGHT_LATEST)?
+        };
+
+        // Fetch quote policy from the consensus layer using the given or the active version.
+        let registry_state = RegistryState::new(&consensus_state);
+        let runtime = registry_state
+            .runtime(Context::create_child(&ctx), runtime_id)?
+            .ok_or(PolicyVerifierError::MissingRuntimeDescriptor)?;
+
+        let ad = match version {
+            Some(version) => runtime
+                .deployment_for_version(version)
+                .ok_or(PolicyVerifierError::NoDeployment)?,
+            None => {
+                let beacon_state = BeaconState::new(&consensus_state);
+                let epoch = beacon_state.epoch(Context::create_child(&ctx))?;
+
+                runtime
+                    .active_deployment(epoch)
+                    .ok_or(PolicyVerifierError::NoDeployment)?
+            }
+        };
+
+        let policy = match runtime.tee_hardware {
+            TEEHardware::TEEHardwareIntelSGX => {
+                let sc: SGXConstraints = ad
+                    .try_decode_tee()
+                    .map_err(|_| PolicyVerifierError::BadTEEConstraints)?;
+                sc.policy()
+            }
+            _ => bail!(PolicyVerifierError::HardwareMismatch),
+        };
+
+        Ok(policy)
+    }
+
+    /// Verify that runtime's quote policy has been published in the consensus layer.
+    pub fn verify_quote_policy(
+        &self,
+        ctx: Arc<Context>,
+        policy_raw: &[u8],
+        runtime_id: &Namespace,
+        version: Option<Version>,
+        use_latest_state: bool,
+    ) -> Result<QuotePolicy> {
+        let untrusted_policy: QuotePolicy = cbor::from_slice(policy_raw)?;
+        let published_policy = self.quote_policy(ctx, runtime_id, version, use_latest_state)?;
+
+        if untrusted_policy != published_policy {
+            debug!(
+                self.logger,
+                "quote policy mismatch, untrusted: {:?}, published: {:?}",
+                untrusted_policy,
+                published_policy
+            );
+            return Err(PolicyVerifierError::PolicyNotPublished.into());
+        }
+
+        Ok(published_policy)
+    }
+
+    /// Fetch key manager's SGX policy from the latest verified consensus layer state.
+    pub fn sgx_policy(
```

Comment:
Key manager has 2 policies, a "key manager policy" and a "key manager quote policy". I used the name `sgx_policy` as struct is called `(Signed)PolicySGX` and `quote_policy` because of `QuotePolicy`.